compileTestJava fails with module split package errors

**Describe the bug**

When trying to compile ical4j with `./gradlew clean check`, I get:

```
> Task :compileTestJava FAILED
warning: [options] module name in --add-exports option not found: org.junit.jupiter.engine
error: the unnamed module reads package org.codehaus.groovy.runtime from both org.codehaus.groovy.test and org.codehaus.groovy
error: the unnamed module reads package groovy.util from both org.codehaus.groovy.test and org.codehaus.groovy
error: the unnamed module reads package groovy.transform from both org.codehaus.groovy.test and org.codehaus.groovy
error: the unnamed module reads package groovy.lang from both org.codehaus.groovy.test and org.codehaus.groovy
error: the unnamed module reads package org.hamcrest from both hamcrest.core and org.hamcrest
error: the unnamed module reads package org.hamcrest.internal from both hamcrest.core and org.hamcrest
error: the unnamed module reads package org.hamcrest.core from both hamcrest.core and org.hamcrest
error: module org.spockframework.core reads package org.hamcrest from both org.hamcrest and hamcrest.core
```
... and a lot more, and finally:
```
100 errors
1 warning

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
```

Looking with `--info`, the `compiler args for task compileTestJava` are identical (just ordered differently). Also the md5sums of the dependencies are equal.

**To Reproduce**

I see CI builds fine, so this is likely a problem with my development setup, but I haven't figured out what yet, and thought I'd share here.

**Environment (please complete the following information):**
 - OS: NixOS Linux
 - Java Version 11
 - iCal4j Version `develop`

--- a/build.gradle
+++ b/build.gradle
@@ -96,6 +96,8 @@ dependencies {
     testImplementation platform("org.spockframework:spock-bom:$spockVersion"),
             "org.spockframework:spock-core"
 
+    testImplementation "org.hamcrest:hamcrest-core:$hamcrestVersion"
+
     // testcontainers
     testImplementation "org.testcontainers:testcontainers:$testcontainersVersion",
             "org.testcontainers:spock:$testcontainersVersion"

--- a/gradle.properties
+++ b/gradle.properties
@@ -8,13 +8,17 @@ threetenExtraVersion = 1.8.0
 jparsecVersion = 3.1
 jsonSkemaVersion = 0.11.0
 
-groovyVersion = 3.0.21
+groovyVersion = 3.0.22
 bndVersion = 6.3.1
 tagsoupVersion = 1.2.1
 jcacheVersion = 2.6.1.Final
 #junitVersion = 4.13
 junitVintageVersion = 5.10.2
-spockVersion = 2.4-M1-groovy-3.0
+# spock pulls in hamcrest 2.2, but junit pulls in
+# hamcrest-core 1.3, so explicitly load hamcrest-core 2.2
+# to make versions match
+hamcrestVersion = 2.2
+spockVersion = 2.4-M4-groovy-3.0
 testcontainersVersion = 1.19.8
 
 jacoco_htmlReport = true

